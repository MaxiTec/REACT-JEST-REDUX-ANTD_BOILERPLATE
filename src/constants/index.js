// Exportamos todos las constantes
export * from './login-types';
export * from './actions-types';
export * from './alert-types';