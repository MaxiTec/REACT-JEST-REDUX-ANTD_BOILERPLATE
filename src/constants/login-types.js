/*
En este archivo exportamos las constantes de las operaciones para poder comunicar
Los Actions con los Reducers
*/
const LOGIN_REQUEST = 'LOGIN_REQUEST';
const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
const LOGIN_ERROR = 'LOGIN_ERROR';
const LOGOUT = 'LOGOUT';

export const loginTypes = {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  LOGOUT,
};
