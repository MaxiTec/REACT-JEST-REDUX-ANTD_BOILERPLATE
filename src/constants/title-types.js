const CHANGE_TITLE = 'CHANGE_TITLE';

export const loginTypes = {
  CHANGE_TITLE,
};
