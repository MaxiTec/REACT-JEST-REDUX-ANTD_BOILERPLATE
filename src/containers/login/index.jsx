import React from 'react';
import { Formik } from 'formik';
import { Spin } from 'antd';
import moment from 'moment';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import * as Yup from 'yup';
import DisplayBookingForm from '../../components/LoginForm';
import { userActions } from '../../actions/user-actions';

const timeFormat = 'h:mm A';
const initialValues = {
  username: '',
  password: '',
};
const SignupSchema = Yup.object().shape({
  username: Yup.string().required('Requerido'),
  password: Yup.string().required('Requerido'),
});
class LoginPage extends React.Component {
  constructor(props) {
    super(props);
    // reset login status, just in Case :)
    // this.props.dispatch(userActions.logout());
    this.state = {
      username: '',
      password: '',
    };
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(formProps, actions) {
    const {
      username, password
    } = formProps;
    console.log( username, password);
    const { dispatch } = this.props;
    dispatch(userActions.login(username, password))
      .then((ele) => {
        console.log('Paso el Login');
      })
      .catch((error) => {
        console.log(error)
        // this.props.form.setFields({
        //   username: {
        //     errors: [new Error('forbid ha')],
        //   },
        // });
      });
  }
  // handleSubmit(e) {
  //   console.log(e);
  //   e.preventDefault();
  //   this.props.form.validateFields((err, values) => {
  //     if (!err) {
  //       const { dispatch } = this.props;
  //       dispatch(userActions.login(values.username, values.password))
  //         .then((ele) => {
  //           console.log('Paso el Login');
  //         })
  //         .catch((ele) => {
  //           console.log(ele);
  //           this.props.form.setFields({
  //             username: {
  //               // value: values.username,
  //               errors: [new Error('forbid ha')],
  //             },
  //           });
  //         });
  //       console.log('Received values of form: ', values);
  //     } else {
  //       console.log(err);
  //     }
  //   });
  // }
  render() {
    const { loggingIn } = this.props;
    console.log(this.props);
    return (
      <div className="col-md-6 col-md-offset-3">
        <h2>Login</h2>
        <Spin spinning={loggingIn} tip="Cargando...">
          <Formik
            initialValues={initialValues}
            validationSchema={SignupSchema}
            onSubmit={this.handleSubmit}
            render={DisplayBookingForm}
          />
        </Spin>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const { loggingIn } = state.login;
  return {
    loggingIn,
  };
}
const connectedLoginPage = withRouter(connect(mapStateToProps)(LoginPage));
export { connectedLoginPage as LoginPage };
