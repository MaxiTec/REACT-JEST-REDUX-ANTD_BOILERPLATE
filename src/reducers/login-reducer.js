import { loginTypes } from '../constants';

const INITIAL_STATE = { loggingIn: false, loggedIn: false };
const login = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case loginTypes.LOGIN_REQUEST:
      return {
        loggingIn: true,
        user: action.user,
      };
    case loginTypes.LOGIN_SUCCESS:
      return {
        loggedIn: true,
        user: action.user,
      };
    case loginTypes.LOGIN_ERROR:
      return INITIAL_STATE;
    case loginTypes.LOGOUT:
      return INITIAL_STATE;
    default:
      return state;
  }
};

export default login;
