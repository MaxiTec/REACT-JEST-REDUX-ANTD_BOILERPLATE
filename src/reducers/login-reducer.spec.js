import login from './login-reducer';
import {loginTypes} from '../constants/login-types'
import {INITIAL_STATE} from './login-reducer'
describe('Reducers Login', () => {
    it('Debe Despachar LOGIN_REQUEST', () => {
        // expect(login(INITIAL_STATE,{type: loginTypes.LOGIN_REQUEST})).toMatchSnapshot()
        expect(login(INITIAL_STATE,{type: loginTypes.LOGIN_REQUEST})).toEqual({"loggingIn": true, "user": undefined});
    })
    // it('Debe Despachar LOGIN_SUCCESS', () => {
    //     const successAction = {
    //         type: loginTypes.LOGIN_SUCCESS,
    //         payload: 'test',
    //       };
    //       expect(login(INITIAL_STATE,successAction)).toEqual({"loggingIn": true, "user": undefined});
    // })
});
