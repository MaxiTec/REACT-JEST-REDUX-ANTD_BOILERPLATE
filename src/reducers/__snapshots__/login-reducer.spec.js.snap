// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers Login Debe Despachar LOGIN_SUCCESS 1`] = `Object {}`;
